dvc/compare.py

```diff
@@ -126,9 +137,13 @@ def shape(self) -> Tuple[int, int]:
         return len(self.columns), len(self)
 
     def drop(self, *col_names: str) -> None:
-        for col_name in col_names:
-            self._keys.remove(col_name)
-            self._columns.pop(col_name)
+        to_remove = set()
+        for col in col_names:
+            if not self.is_protected(col):
+                to_remove.add(col)
```

Comment:
How about remove `to_remove` and directly:
```python
        for col in col_names:
            if not self.is_protected(col):
                self._keys.remove(col)
                self._columns.pop(col)

```

Reply:
It merged automatically.

Reply:
#7299